cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MigrationTableUtility.java

```diff
@@ -0,0 +1,129 @@
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * To get v2/v3 metrics tables; perform tableExists/delete table on v2 metrics tables
+ */
+public class MigrationTableUtility {
+  private static final Logger LOG = LoggerFactory.getLogger(MigrationTableUtility.class);
+
+  private final CConfiguration cConf;
+  private final Configuration hConf;
+  private final MetricDatasetFactory metricDatasetFactory;
+  private final HBaseDDLExecutorFactory hBaseDDLExecutorFactory;
+  private final HBaseTableUtil tableUtil;
+
+  MigrationTableUtility(CConfiguration cConfiguration, Configuration hConf,
+                        MetricDatasetFactory metricDatasetFactory, HBaseTableUtil tableUtil) {
+    this.cConf = cConfiguration;
+    this.hConf = hConf;
+    this.metricDatasetFactory = metricDatasetFactory;
+    this.hBaseDDLExecutorFactory = new HBaseDDLExecutorFactory(cConf, hConf);
+    this.tableUtil = tableUtil;
+  }
+
+  // we try to get v2 metrics table; if there is any exception while getting the table we return null
+  @Nullable
+  MetricsTable getV2MetricsTable(int resolution) {
+    MetricsTable v2table = null;
+    try {
+      v2table = metricDatasetFactory.getV2MetricsTable(resolution);
+    } catch (Exception e) {
+      LOG.info("Cannot access v2 metricsTable due to exception", e);
+    }
+    return v2table;
+  }
+
+  // we try to get v3 metrics table; if there is any exception while getting the table we return null
+  @Nullable
+  MetricsTable getV3MetricsTable(int resolution) {
+    MetricsTable v3table = null;
+    try {
+      v3table = metricDatasetFactory.getV3MetricsTable(resolution);
+    } catch (Exception e) {
+      LOG.info("Cannot access v3 metricsTable due to exception", e);
+    }
+    return v3table;
+  }
+
+  /**
+   * check if v2 metrics table exists for the resolution
+   * @return true if table exists; false otherwise
+   */
+  boolean v2MetricsTableExists(int resolution) {
+    TableId tableId = getV2MetricsTableName(resolution);
+    try {
+      try (HBaseAdmin admin = new HBaseAdmin(hConf)) {
+        TableId hBaseTableId =
+          tableUtil.createHTableId(new NamespaceId(tableId.getNamespace()), tableId.getTableName());
+        boolean doesExist  = tableUtil.tableExists(admin, hBaseTableId);
+        LOG.debug("Table {} exists : {}", hBaseTableId.getTableName(), doesExist);
+        return doesExist;
+      }
+    } catch (IOException e) {
+      LOG.info("Exception while checking table exists", e);
+    }
+    return false;
+  }
+
+  /**
+   * delete v2 metrics table for the resolution
+   * @param resolution resolution of metrics table to delete
+   * @return true if deletion is successful; false otherwise
+   */
+  boolean deleteV2MetricsTable(int resolution) {
+    TableId tableId = getV2MetricsTableName(resolution);
+    try {
+      try (HBaseDDLExecutor ddlExecutor = hBaseDDLExecutorFactory.get(); HBaseAdmin admin = new HBaseAdmin(hConf)) {
```

Comment:
You should use the dataset framework to delete the table. Otherwise it will still exist as a dataset.